Modules/debug.py

```diff
@@ -47,16 +47,16 @@ def text_tab(text, tab):
     return space_number * space + text
 
 def list_to_string(item_in_list, current_indentation=0):
-    statement = ""
+    statement = DebugCore.NEWLINE
     for item in item_in_list:
         statement = statement + text_tab(item, current_indentation + 1) + DebugCore.NEWLINE
     return statement
 
 def dict_to_string(dictionary, current_indentation=0):
-    statement = ""
-    for key, value in enumerable(dictionary):
+    statement = DebugCore.NEWLINE
+    for key, value in dictionary.items():
         item = "%s:%s" % (key, value)
-        statement = statement + (item, current_indentation + 1) + DebugCore.NEWLINE
+        statement = statement + text_tab(item, current_indentation + 1) + DebugCore.NEWLINE
```

Comment:
we really need to get around to writing debugging methods for these. I didn't even know it was broken until I tried to call it.